docs/testing.rst

```diff
@@ -0,0 +1,36 @@
+Testing Details
```

Comment:
Maybe something like "Test Environment Configuration" would be more accurate?  Otherwise, looks great.